File: backend/src/main/java/z9/hobby/domain/favorite/entity/FavoriteEntity.java
package z9.hobby.domain.favorite.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "favorites")
public class FavoriteEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "favorite_id")
    private Long id;

    @Column(name = "favorite_name", nullable = false)
    private String name;

    public static FavoriteEntity createNewFavorite(String name) {
        return FavoriteEntity.builder()
                .name(name)
                .build();
    }
}

File: backend/src/main/java/z9/hobby/domain/favorite/entity/FavoriteEntity.kt
package z9.hobby.domain.favorite.entity;

import jakarta.persistence.*

@Entity
@Table(name = "favorites")
class FavoriteEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "favorite_id")
    val id: Long? = null,

    @Column(name = "favorite_name", nullable = false)
    var name: String
) {
    companion object {
        @JvmStatic
        fun createNewFavorite(name: String): FavoriteEntity {
            return FavoriteEntity(name = name)
        }
    }
}
